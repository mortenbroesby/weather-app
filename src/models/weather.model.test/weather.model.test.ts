import { WeatherModel } from "../weather.model";

describe("models/weather.model.ts", () => {
  it("Generates a valid Weather model", () => {
    const SampleJson = require("./weather.example.valid.richData");
    const model = new WeatherModel(SampleJson);

    expect(model).toMatchSnapshot();
  });

  it("Generates a valid Weather model with poor data", () => {
    const SampleJson = require("./weather.example.valid.poorData");
    const model = new WeatherModel(SampleJson);

    expect(model).toMatchSnapshot();
  });

  it("Generates a valid Weather model with main data missing", () => {
    const SampleJson = require("./weather.example.valid.noMain");
    const model = new WeatherModel(SampleJson);

    expect(model).toMatchSnapshot();
  });

  it("Generates a valid Weather model on malformed data", () => {
    const model = new WeatherModel();

    expect(model).toMatchSnapshot();
  });
});
