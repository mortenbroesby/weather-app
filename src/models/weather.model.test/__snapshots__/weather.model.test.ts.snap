// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`models/weather.model.ts Generates a valid Weather model 1`] = `
WeatherModel {
  "humidity": 68,
  "precipitation": 10,
  "temperature": Object {
    "current": 19,
    "maximum": 20,
    "minimum": 18,
  },
  "timestamp": 1566142435,
  "weatherType": Object {
    "description": "shower rain",
    "id": 521,
    "type": "Rain",
  },
  "windSpeed": 7,
}
`;

exports[`models/weather.model.ts Generates a valid Weather model on malformed data 1`] = `
WeatherModel {
  "humidity": 0,
  "precipitation": 0,
  "temperature": Object {
    "current": 0,
    "maximum": 0,
    "minimum": 0,
  },
  "timestamp": 1566227684248,
  "weatherType": Object {
    "description": "clear sky",
    "id": 800,
    "type": "Clear",
  },
  "windSpeed": 0,
}
`;

exports[`models/weather.model.ts Generates a valid Weather model with main data missing 1`] = `
WeatherModel {
  "humidity": 0,
  "precipitation": 0,
  "temperature": Object {
    "current": 0,
    "maximum": 0,
    "minimum": 0,
  },
  "timestamp": 1566227684248,
  "weatherType": Object {
    "description": "clear sky",
    "id": 800,
    "type": "Clear",
  },
  "windSpeed": 0,
}
`;

exports[`models/weather.model.ts Generates a valid Weather model with poor data 1`] = `
WeatherModel {
  "humidity": 0,
  "precipitation": 0,
  "temperature": Object {
    "current": 0,
    "maximum": 0,
    "minimum": 0,
  },
  "timestamp": 1566227684244,
  "weatherType": Object {
    "description": "clear sky",
    "id": 800,
    "type": "Clear",
  },
  "windSpeed": 0,
}
`;
